File: src/main/java/co/onmind/microhex/infrastructure/controllers/HealthController.java
package co.onmind.microhex.infrastructure.controllers;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

/**
 * Simple health check controller for testing.
 */
@RestController
@RequestMapping("/api")
public class HealthController {
    
    @GetMapping("/health")
    public ResponseEntity<Map<String, String>> health() {
        return ResponseEntity.ok(Map.of(
            "status", "UP",
            "application", "hex4j",
            "version", "2.0.0"
        ));
    }
}

File: src/main/kotlin/co/onmind/hex/infrastructure/controllers/HealthController.kt
package co.onmind.hex.infrastructure.web

import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

/**
 * Simple health check controller to test if routing is working.
 */
@Controller("/health")
class HealthController {
    
    @Get
    fun health(): HttpResponse<Map<String, String>> {
        return HttpResponse.ok(mapOf("status" to "UP"))
    }
}